UITests/DomainAutocompleteTests.swift

```diff
@@ -21,6 +21,14 @@ class DomainAutocompleteTests: KIFTestCase {
         ensureAutocompletionResult(textField, prefix: "www.yahoo.com/", completion: "")
         tester().clearTextFromFirstResponder()
 
+        tester().enterTextIntoCurrentFirstResponder("www.yah")
+        ensureAutocompletionResult(textField, prefix: "www.yah", completion: "oo.com/")
+        tester().enterTextIntoCurrentFirstResponder("\u{0008}")
+        ensureAutocompletionResult(textField, prefix: "www.yah", completion: "")
+        tester().enterTextIntoCurrentFirstResponder("\u{0008}")
+        ensureAutocompletionResult(textField, prefix: "www.ya", completion: "")
+        tester().clearTextFromFirstResponder()
```

Comment:
I'd type "h" again at this point to confirm that entering more characters after some deletions still works.

We could also use a check that deletes all of the characters, then makes sure the search controller disappears.